### PR TITLE
Edit User Screen (with ID implemented)

### PR DESCRIPTION
**Pull request by thongquach-eh**
- Finished Edit User Screen
- Implement ID for all the screens
- The following video is the demo for Add User Screen and Edit User Screen (0:38) with ID implemented:

https://user-images.githubusercontent.com/81946174/116027570-70e44e80-a67f-11eb-9fdd-bdefbf7a9861.mov

### CODE REVIEW

File: src/components/UserListScreen.js [hunk 1]
```diff
@@ -32,7 +32,6 @@ const UserListScreen = ({
       renderItem={({item: user}) => (
         <UserItem user={user} navigation={navigation} />
       )}
-      keyExtractor={user => user.email}
```
Comment (ttkien) [inline src/components/UserListScreen.js:35]: Flatlist should hve keyExtractor to improve performance

Comment (thongquach-eh) [reply src/components/UserListScreen.js:35]: Sorry, I made a mistake here. I thought id would be by default use as key. I have fixed it.

File: src/components/AddUserScreen.js [hunk 2]
```diff
@@ -125,104 +143,114 @@ const AddUserScreen = ({navigation, route}: AddUserScreenProps): Node => {
...
+            label="Postcode"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.postcode', user)
+            }
+          />
+          <UserInput
+            label="Country"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.country', user)
+            }
```
Comment (ttkien) [inline src/components/AddUserScreen.js]: These component are duplicated with EditUserScreen. Please try to separate them into a component

Comment (ttkien) [reply src/components/AddUserScreen.js]: And treat above components Gender, Birthday,... as the same

Comment (thongquach-eh) [reply src/components/AddUserScreen.js]: Thank you for your feedback. I've refactored and split them into separate reusable components

File: src/components/TitlePicker.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+});
+
+const TitlePicker = ({
+  label,
+  value,
+  onValueChange,
+}: {
+  label: string,
+  value: ?string,
+  onValueChange: ?(itemValue: string | number, itemIndex: number) => mixed,
```
Comment (ttkien) [inline src/components/TitlePicker.js]: It's minor that `itemValue` => should be `value`, `itemIndex` => should be `index`. You don't need to change this, just remember

File: src/components/LocationInput.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    marginVertical: 10,
+  },
+  container: {
+    marginLeft: 10,
+  },
+});
+
+const LocationInput = ({
+  label,
+  user,
```
Comment (ttkien) [inline src/components/LocationInput.js]: It's about Component Design. LocationInput don't need to know `user` object. It only needs to know `location` object. and function `writeValue` should be function `onChangeLocation`

File: src/components/GenderPicker.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+});
+
+const GenderPicker = ({
+  label,
+  value,
+  onValueChange,
+}: {
+  label: string,
+  value: ?string,
+  onValueChange: ?(itemValue: string | number, itemIndex: number) => mixed,
```
Comment (ttkien) [inline src/components/GenderPicker.js]: Could you explain why you return `=> mixed`, why don't use `=> void`

File: src/components/FullNameInput.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+  },
+});
+
+const FullNameInput = ({
+  user,
```
Comment (ttkien) [inline src/components/FullNameInput.js]: It's about Component Design. FullNameInput don't need to know user object. It only needs to know First name, Last name, Title object. and function writeValue should be function onChangeLocation

File: src/components/GenderPicker.js (hunk 5 above)
Comment (thongquach-eh) [reply src/components/GenderPicker.js]: From my perspective, this component is a wrapper of Picker component. Hence, I took the signature of the onValueChange prop of Picker component, and put it here

File: src/components/TitlePicker.js (hunk 3 above)
Comment (thongquach-eh) [reply src/components/TitlePicker.js]: Thank you for your feedback. The names were taken from the Picker component itself, but I think your argument makes much more sense. Hence, I have changed the parameters name accordingly.

File: src/components/LocationInput.js (hunk 4 above)
Comment (thongquach-eh) [reply src/components/LocationInput.js]: Thank you for your feedback. I have rewritten the component accordingly.

File: src/components/FullNameInput.js (hunk 6 above)
Comment (thongquach-eh) [reply src/components/FullNameInput.js]: I have rewritten the component accordingly.

**Review by ttkien — APPROVED** [review]
LGTM

**Pull request merged**